### PR TITLE
Normalize metadata

### PR DESCRIPTION
**Pull request by choccccy**
from #79:

> It's been annoying me for a while that each of the metadata fields for our vector DB are a bit different:
> 
> The ones in `pgvector_data_doc` use:
> 
> ```sql
>     tags TEXT[]                             -- tags associated with the chunk
> ```
> 
> And `pgvector_message` uses:
> 
> ```sql
>     metadata JSON                             -- additional metadata of the message
> ```
> 
> I think this second pattern is the one we should be using, so in effect it would just be an update to `ogbujipt.embedding.pgvector_data_doc`.

This change from lists of strings, to (JSON) dictionaries, is the whole branch.

This did come with the loss of tag based searching; it is expected that the user would implement this themselves, instead.

### CODE REVIEW

**Review by olajiide — APPROVED** [review]

File: pylib/__about__.py
Comment (uogbuji) [inline pylib/__about__.py:1]: 0.9.0 hasn't been released, so we should revert this change, but AFTER today 😆

Comment (choccccy) [reply pylib/__about__.py:1]: This was to avoid a conflict between your working 0.9.0 stuff and this branch
